### PR TITLE
Tab bar

### PR DESCRIPTION
**Pull request by fedeGiussa25**
## Summary

- Added Home Tab Bar.

## Screenshots

[Change!] Show the screenshots of the views you modified.

#### Galaxy S (or similar)
Peek 2020-11-05 12-53

#### Pixel 2 (or similar)
Peek 2020-11-05 12-40

#### Pixel 3 or 3XL (or similar)
Peek 2020-11-05 12-47

### iOS
- Not for this training.

## Trello Card
https://trello.com/c/qvxPMvhC/18-n-4-tab-bar

### CODE REVIEW

File: src/app/screens/Home/index.tsx [hunk 1]
```diff
@@ -1,15 +1,24 @@
 import React from 'react';
-import { View } from 'react-native';
-import BookList from '@app/components/BookList';
-import { BOOK_LIST } from '@constants/books';
+import { Image } from 'react-native';
+import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
+import Library from '@app/screens/Library';
+import Rentals from '@app/screens/Rentals';
```
Comment (mattgle) [inline src/app/screens/Home/index.tsx:5]:
```suggestion
import Library from '@screens/Library';
import Rentals from '@screens/Rentals';
```

File: src/app/screens/Library/index.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { View } from 'react-native';
+import BookList from '@app/components/BookList';
```
Comment (mattgle) [inline src/app/screens/Library/index.tsx:3]:
```suggestion
import BookList from '@components/BookList';
```

File: src/app/screens/Home/index.tsx [hunk 3]
```diff
@@ -1,15 +1,24 @@
...
-    </View>
+    <Tab.Navigator
+      screenOptions={({ route }) => ({
+        tabBarIcon: ({ focused }) => (
+          <Image source={getIcon(ROUTES_ICONS[route.name], focused)} resizeMode="contain" />
+        )
+      })}>
+      <Tab.Screen name={Routes.Library} component={Library} options={{ title: Routes.Library }} />
+      <Tab.Screen name={Routes.Rentals} component={Rentals} options={{ title: Routes.Rentals }} />
+    </Tab.Navigator>
```
Comment (mattgle) [inline src/app/screens/Home/index.tsx:21]: i think this logic should be at the App component

File: src/app/screens/Library/index.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import { BOOK_LIST } from '@constants/books';
+
+import styles from './styles';
+
+function Library() {
+  return (
+    <View style={styles.container}>
+      <BookList data={BOOK_LIST} />
+    </View>
+  );
```
Comment (mattgle) [inline src/app/screens/Library/index.tsx:13]: is this view necessary? Just asking

File: src/constants/routes.ts [hunk 5]
```diff
@@ -1,6 +1,8 @@
 enum Routes {
-  Home = 'Library',
-  Detail = 'Book Detail'
+  Home = 'Home',
+  Library = 'Library',
+  Detail = 'Book Detail',
+  Rentals = 'Rentals'
```
Comment (mattgle) [inline src/constants/routes.ts:5]: Routes and Screens names should match

**Review by mattgle — CHANGES_REQUESTED** [review]
just a few comments! Great work btw

File: src/app/screens/Library/index.tsx (hunk 4 above)
Comment (fedeGiussa25) [reply src/app/screens/Library/index.tsx]: I checked and it's necessary. It's so the booklist renders below the header.

File: src/app/screens.tsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+          <Image source={getIcon(ROUTES_ICONS[route.name], focused)} resizeMode="contain" />
+        )
+      })}>
+      <Tab.Screen name={Routes.Library} component={Library} options={{ title: Routes.Library }} />
+      <Tab.Screen name={Routes.Rentals} component={Rentals} options={{ title: Routes.Rentals }} />
+    </Tab.Navigator>
+  );
+}
+
+export default Home;
```
Comment (mattgle) [inline src/app/screens.tsx:25]: this shouldn't be named as Home

File: src/constants/icons.ts [hunk 7]
```diff
@@ -1 +1,24 @@
+import LibraryActiveIcon from '@assets/icLibraryActive.png';
+import LibraryIcon from '@assets/icLibrary.png';
+import MyRentalsActiveIcon from '@assets/icMyRentalsActive.png';
+import MyRentalsIcon from '@assets/icMyRentals.png';
```
Comment (mattgle) [inline src/constants/icons.ts:4]: assets should be in camelCase

File: src/interfaces/tab.ts [hunk 8]
```diff
@@ -0,0 +1,8 @@
+export interface TabIcon {
+  focused: HTMLImageElement;
+  unfocused: HTMLImageElement;
```
Comment (mattgle) [inline src/interfaces/tab.ts:3]: mhm this doesnt look right since we dont have HTML in RN

**Review by mattgle — CHANGES_REQUESTED** [review]
great work some comments!

**Review by mattgle — APPROVED** [review]

**Pull request merged**